documentation/src/main/asciidoc/search-dsl.asciidoc

```diff
@@ -31,3 +36,206 @@ include::todo-placeholder.asciidoc[]
+Similarly, it is possible to pass an argument of type `Object` in the sort DSL
+when defining the behavior for missing values (`onMissingValue().use(Object)`).
+
+These methods do not actually accept *any* object,
+and will throw an exception when passed an argument with the wrong type.
+
+Generally the expected type of this argument should be rather obvious:
+for example if you created a field by mapping an `Integer` property,
+then an `Integer` value will be expected when building a predicate;
+if you mapped a `java.time.LocalDate`, then a `java.time.LocalDate` will be expected,
+etc.
+
+Things get a little more complex if you start defining and using custom bridges.
+You will then have properties of type `A` mapped to an index field of type `B`.
+What should you pass to the DSL?
+To answer that question, we need to understand DSL converters.
+
+DSL converters are a feature of Hibernate Search that allows the DSL to accept
+arguments that match the type of the indexed property,
+instead of the type of the underlying index field.
+
+Each custom bridge has the possibility to define a DSL converter for the index fields it populates.
+When it does, every time that field is mentioned in the predicate DSL,
+Hibernate Search will use that DSL converter to convert the value passed to the DSL to a value that the backend understands.
+
+For example, let's imagine an `AuthenticationEvent` entity with an `outcome` property of type `AuthenticationOutcome`.
+This `AuthenticationOutcome` type is an enum.
+We index the `AuthenticationEvent` entity and its `outcome` property in order to allow users to find events by their outcome.
+
+The default bridge for enums puts the result of `Enum.name()` into a `String` field.
+However, this default bridge also defines a DSL converter.
+As a result, any call to the DSL will be expected to pass an `AuthenticationOutcome` instance:
+
+.Transparent conversion of DSL parameters
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/search/documentation/searchdsl/DslConverterIT.java[tags=dsl-converter-enabled]
+----
+====
+
+This is handy, and especially appropriate if users are asked to select an outcome in a list of choices.
+But what if we want users to type in some words instead, i.e. what if we want full-text search on the `outcome` field?
+Then we will not have an `AuthenticationOutcome` instance to pass to the DSL, only a `String`...
+
+In that case, we will first need to assign some text to each enum.
+This can be achieved by defining a custom <<mapper-orm-bridge-valuebridge,`ValueBridge<AuthenticationOutcome, String>`>>
+and applying it to the `outcome` property so as to index a textual description of the outcome,
+instead of the default `Enum#name()`.
+
+Then, we will need to tell Hibernate Search that the value passed to the DSL should not be passed to the DSL converter,
+but should be assumed to match the type of the field (in this case, `String`).
```

Comment:
It may not be stored at all (just indexed) and even then it's not stored as a String, but as `BytesRef`. So this proposal is not ideal. But I'll change it to be a bit more specific.

Reply:
thanks